src/containers/Affiliate/index.tsx

```diff
@@ -0,0 +1,141 @@
+import CopyIcon from '~/components/Icons/CopyIcon'
+
+const Affiliate = () => {
+    const { account } = useActiveWeb3React()
+    const { createAffiliateCode, getAffiliateCode } = useFuulSDK()
+    const [affiliateCode, setAffiliateCode] = useState<string>('')
+    const [newAffiliateCode, setNewAffiliateCode] = useState<string>('')
+    const [error, setError] = useState<string | null>(null)
+    const [copied, setCopied] = useState(false)
+    const [hasFetched, setHasFetched] = useState<boolean>(false)
+
+    useEffect(() => {
+        if (account && !hasFetched) {
+            setHasFetched(true)
+            ;(async () => {
+                try {
+                    const code = await getAffiliateCode(account)
+                    if (code) {
+                        setAffiliateCode(code)
+                    }
+                } catch (err) {
+                    console.error('Error fetching affiliate code:', err)
+                }
+            })()
+        }
+    }, [account, getAffiliateCode, hasFetched])
+
+    const handleCreateAffiliateCode = async () => {
+        if (account && newAffiliateCode) {
+            try {
+                await createAffiliateCode(account, newAffiliateCode)
+                setAffiliateCode(newAffiliateCode)
+                setNewAffiliateCode('')
+                setError(null)
+            } catch (err) {
+                console.error('Error creating affiliate code:', err)
+                setError((err as Error)?.message || 'Failed to create affiliate code. Please try again.')
+            }
+        }
+    }
+
+    return (
+        <Container>
+            <Content>
+                <Title>Affiliate Program</Title>
+                {affiliateCode ? (
+                    <>
+                        <AffiliateText>
+                            Your Affiliate Code: <BoldText>{affiliateCode}</BoldText>
+                        </AffiliateText>
+                        <AffiliateText>
+                            Your Affiliate Link: <BoldText>{`https://app.opendollar.com?af=${affiliateCode}`}</BoldText>
+                        </AffiliateText>
+                        <CopyToClipboard
+                            text={`https://app.opendollar.com?af=${affiliateCode}`}
+                            onCopy={() => {
+                                setCopied(true)
+                                setTimeout(() => setCopied(false), 1500)
+                            }}
+                        >
+                            <Button>
+                                {copied ? 'Copied!' : 'Copy Affiliate Link'}
+                                <CopyIconContainer>
+                                    <CopyIcon />
+                                </CopyIconContainer>
+                            </Button>
+                        </CopyToClipboard>
+                    </>
+                ) : (
+                    <>
+                        <input
```

Comment:
Can you show the url on the left, and the input box on the right? This way its clear how the final result will look.

Here's how I imagine it

![image](https://github.com/open-dollar/od-app/assets/35622595/5bf5e215-6df8-4db4-8ff6-b87af4d66541)